app/components/primer/octicon_component.rb

```diff
@@ -38,7 +39,7 @@ def initialize(icon:, size: SIZE_DEFAULT, **system_arguments)
       # Filter out classify options to prevent them from becoming invalid html attributes.
       # Note height and width are both classify options and valid html attributes.
       octicon_helper_options = @system_arguments.slice(:height, :width)
-      @system_arguments = @system_arguments.except(*Primer::Classify::VALID_KEYS, :classes).merge(octicon_helper_options)
```

Comment:
Could you help me understand why this is an issue only with octicon?

Reply:
because octicon does not use `Primer::BaseComponent`

Reply:
aha! Why do we do that?

Reply:
because we want to use the `OcticonHelper` to handle the `svg`generation

Reply:
Why can't we include that helper were we to inherit from BaseComponent instead?